apps/frontend/src/hooks/useAaveReservesData.tsx

```diff
@@ -0,0 +1,67 @@
+import {
+  ReserveDataHumanized,
+  ReservesDataHumanized,
+  UiPoolDataProvider,
+} from '@aave/contract-helpers';
+import { formatReserves, FormatReserveUSDResponse } from '@aave/math-utils';
+
+import { useEffect, useMemo, useState } from 'react';
+
+import dayjs from 'dayjs';
+
+import { config } from '../constants/aave';
+
+export const useAaveReservesData = () => {
+  const provider = config.provider; // TODO: replace with useAccount
+  const [reserves, setReserves] = useState<
+    (ReserveDataHumanized & FormatReserveUSDResponse)[]
+  >([]);
+  const [reservesData, setReservesData] =
+    useState<ReservesDataHumanized | null>(null);
+
+  const uiPoolDataProvider = useMemo(
+    () =>
+      provider
+        ? new UiPoolDataProvider({
+            provider,
+            uiPoolDataProviderAddress: config.UiPoolDataProviderV3Address,
+            chainId: config.chainId,
+          })
+        : null,
+    [provider],
+  );
+
+  useEffect(() => {
+    const fetchReservesData = async () => {
```

Comment:
Would it be possible to move the fetchReservesData function outside of the useEffect to avoid recreating the function on every run? We could then pass uiPoolDataProvider as a parameter to the function. This change might help in optimizing our component's performance by reducing the function's declarations and ensuring it's only defined once.